invenio_records_resources/records/api.py

```diff
@@ -63,11 +63,11 @@ def get_by_key(cls, record_id, key):
                 return cls(obj.data, model=obj)
 
     @classmethod
-    def list_by_record(cls, record_id, with_deleted=False):
+    def list_by_record(cls, record_id, include_deleted=False):
         """List all record files by record ID."""
         query = cls.model_cls.query.filter(cls.model_cls.record_id == record_id)
 
-        if not with_deleted:
+        if not include_deleted:
```

Comment:
query below applies for records which were deleted via .delete method of invenio-records api.
the records we delete are deleted by .delete_record of invenio-rdm-records, this query will cause issues, probably if you delete records in a new way, the files still show up on the list